src/hotspot/share/gc/shenandoah/shenandoahThreadLocalData.hpp

```diff
@@ -50,6 +52,7 @@ class ShenandoahThreadLocalData {
   bool _force_satb_flush;
   int  _disarmed_value;
   double _paced_time;
+  ShenandoahMarkRefsSuperClosure* _mark_closure;
```

Comment:
This rubs me the wrong way. Closures are usually stack-allocated, so we are exposing the stack pointer here.

Reply:
Yeah we need to pass it between the mark-loop and the reference-processor. It's still thread-local.

Reply:
While I am browsing the code here... why not record it in `ShenandoahRefProcThreadLocal` then? It has a benefit of not polluting the "common" `ShenandoahThreadLocalData`, and clearly says the whole thing is about Shenandoah reference processor.